modules/core/src/test/java/org/apache/ignite/internal/processors/cache/expiry/IgniteCacheExpiryPolicyAbstractTest.java

```diff
@@ -1174,9 +1189,18 @@ public void testTouch() throws Exception {
         for (final Integer key : keys) {
             log.info("Update duration on access, key: " + key);
 
-            boolean updated = cache.touch(key);
+            if (async) {
+                IgniteFuture<Boolean> fut = cache.touchAsync(key);
+
+                assertTrue("touchAsync was not completed in 1 sec.", waitForCondition(() -> fut.isDone(), 1_000));
```

Comment:
Why not `IgniteFuture#get(long, java.util.concurrent.TimeUnit)`?
By the way, isn't 1 sec too little? Any thread or JVM freeze and the test would become flaky. I think it's safe to put bigger timeout here, assuming that in most cases the future will be completed very fast anyway.

Reply:
This makes sense to me. will change.